examples/c/src/phi3v.cpp

```diff
@@ -12,12 +12,14 @@ void CXX_API(const char* model_path, const char* execution_provider) {
   std::cout << "Creating config..." << std::endl;
   auto config = OgaConfig::Create(model_path);
 
-  config->ClearProviders();
   std::string provider(execution_provider);
-  if (provider.compare("cpu") != 0) {
-    config->AppendProvider(execution_provider);
-    if (provider.compare("cuda") == 0) {
-      config->SetProviderOption(execution_provider, "enable_cuda_graph", "0");
+  if (provider.compare("follow_config") != 0) {
+    config->ClearProviders();
+    if (provider.compare("cpu") != 0) {
+      config->AppendProvider(execution_provider);
+      if (provider.compare("cuda") == 0) {
+        config->SetProviderOption(execution_provider, "enable_cuda_graph", "0");
+      }
```

Comment:
Is it easy to move this into common.cpp? It appears to be the same in every cpp example

Reply:
Yeah I think this one is easy. I can do that.

Reply:
Alright, done -- also moved some of the argument parsing logic to common as well. Take a look when you get a chance.